File: BotsSDK/korebot/src/main/java/kore/botssdk/speech/engine/SpeechRecognitionEngine.java
package kore.botssdk.speech.engine;

import android.content.Context;
import android.speech.RecognitionListener;

import java.util.Locale;

import kore.botssdk.speech.GoogleVoiceTypingDisabledException;
import kore.botssdk.speech.SpeechDelegate;
import kore.botssdk.speech.SpeechRecognitionNotAvailable;
import kore.botssdk.speech.ui.SpeechProgressView;

public interface SpeechRecognitionEngine extends RecognitionListener {

    void init(Context context);

    void clear();

    String getPartialResultsAsString();

    void initSpeechRecognizer(Context context);

    void startListening(SpeechProgressView progressView, SpeechDelegate delegate) throws SpeechRecognitionNotAvailable, GoogleVoiceTypingDisabledException;

    void stopListening();

    void returnPartialResultsAndRecreateSpeechRecognizer();

    void setPartialResults(boolean getPartialResults);

    void shutdown();

    boolean isListening();

    Locale getLocale();

    void setLocale(Locale locale);

    void setPreferOffline(boolean preferOffline);

    void setTransitionMinimumDelay(long milliseconds);

    void setStopListeningAfterInactivity(long milliseconds);

    void setCallingPackage(String callingPackage);

    void unregisterDelegate();
}

File: BotSdk/ui/src/main/java/com/kore/speech/engine/SpeechRecognitionEngine.kt
package com.kore.speech.engine

import android.content.Context
import android.speech.RecognitionListener
import com.kore.speech.GoogleVoiceTypingDisabledException
import com.kore.speech.SpeechDelegate
import com.kore.speech.SpeechRecognitionNotAvailable
import java.util.Locale

interface SpeechRecognitionEngine : RecognitionListener {
    fun init(context: Context)
    fun clear()
    fun getPartialResultsAsString(): String
    fun initSpeechRecognizer(context: Context)

    @Throws(SpeechRecognitionNotAvailable::class, GoogleVoiceTypingDisabledException::class)
    fun startListening(progressView: com.kore.speech.ui.SpeechProgressView?, delegate: SpeechDelegate)
    fun stopListening()
    fun returnPartialResultsAndRecreateSpeechRecognizer()
    fun setPartialResults(getPartialResults: Boolean)
    fun shutdown()
    fun isListening(): Boolean
    fun getLocale(): Locale
    fun setPreferOffline(preferOffline: Boolean)
    fun setTransitionMinimumDelay(milliseconds: Long)
    fun setStopListeningAfterInactivity(milliseconds: Long)
    fun setCallingPackage(callingPackage: String?)
    fun unregisterDelegate()
    fun setLocale(locale: Locale)
}
